test/unit/host_test.rb

```diff
@@ -2641,6 +2641,12 @@ def to_managed!
 
       assert_equal actual_attr['zzz_id'], 1111
     end
+
+    test 'should add inherited attributes when hostgroup in attributes' do
+      hg = FactoryGirl.create(:hostgroup, :with_puppetclass)
+      host = Host.new(:name => "test-host", 'hostgroup' => hg)
```

Comment:
The requirement to use strings for attribute names is also mentioned in the bug report and shows up here.  It's more usual to use symbols from code - can you change it to use an indifferent hash?  Perhaps change the `hash_clone(attributes)` line to add `.with_indifferent_access`.